lib/exp.py

```diff
@@ -30,6 +31,37 @@ def __init__(self, namespace='default'):
         self.k8s = client.CustomObjectsApi()
         self.batch = client.BatchV1Api()
 
+    # Helper methods
+
+    def _retry_poll_api(self, api, max_retries_error, max_retries=30,
+                        retry_interval=1, api_kwargs={}):
+        """
+        Helper function that has a polling loop to retry calling the specified
+        API until it's successful (the client does not throw an Api Exception).
+
+        :param api: Client API to call
+        :param max_retries_error: Error message to print if the maximum retries
+                                  has been reached and the API call still fails
+        :param max_retries: Maximum number of times to retry calling the API
+        :param retry_interval: Number of seconds to wait between API retries
+        :param api_kwargs: Dictionary of API arguments
+        :return: Return value of the client API call
+        """
+        retry_count = 0
+        while retry_count < max_retries:
+            try:
+                return api(**api_kwargs)
+            except client.rest.ApiException:
+                time.sleep(retry_interval)
+                retry_count += 1
+
+                if retry_count >= max_retries:
+                    print(max_retries_error)
+                    raise
+                else:
+                    print("Retrying {}/{} \r".format(retry_count,
```

Comment:
why aren't we using standard python logging for this type of stuff?  I worry that things will be pretty chatty and there's no way to selectively turn off this type of info

Reply:
I'll change it to use python logging